burn-wgpu/src/kernel/comparison.rs

```diff
@@ -0,0 +1,128 @@
+use super::{build_info, KernelSettings, StaticKernel};
+use crate::{context::WorkGroup, element::WgpuElement, kernel_wgsl, tensor::WgpuTensor};
+use burn_tensor::Shape;
+
+kernel_wgsl!(ComparisonRaw, "../template/comparison/binary.wgsl");
+kernel_wgsl!(
+    ComparisonInplaceRaw,
+    "../template/comparison/binary_inplace.wgsl"
+);
+
+#[macro_export]
+macro_rules! comparison {
+    (
+        $struct:ident,
+        $ops:expr
+    ) => {
+        pub struct $struct;
+
+        impl $crate::kernel::StaticKernel for $struct {
+            fn source_template() -> $crate::kernel::SourceTemplate {
+                $crate::kernel::ComparisonRaw::source_template().register(
+                    "body",
+                    format!(
+                        "output[global_id.x] = u32(lhs[index_lhs] {} rhs[index_rhs]);",
+                        $ops
+                    ),
+                )
+            }
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! comparison_inplace {
+    (
+        $struct:ident,
+        $ops:expr
+    ) => {
+        pub struct $struct;
+
+        impl $crate::kernel::StaticKernel for $struct {
+            fn source_template() -> $crate::kernel::SourceTemplate {
+                $crate::kernel::ComparisonInplaceRaw::source_template()
+                    .register(
+                        "body",
+                        "lhs[index_lhs] = compare(lhs[index_lhs], rhs[index_rhs]);",
+                    )
+                    .add_template(format!(
+                        "{}return {{{{ elem }}}}(lhs {} rhs);{}",
+                        "fn compare(lhs: {{ elem }}, rhs: {{ elem }}) -> {{ elem }} {\n",
+                        $ops,
+                        "\n}\n"
+                    ))
+            }
+        }
+    };
+}
+
+pub fn comparison<K: StaticKernel, E: WgpuElement, const D: usize>(
+    lhs: WgpuTensor<E, D>,
+    rhs: WgpuTensor<E, D>,
+) -> WgpuTensor<u32, D> {
```

Comment:
`BoolTensors` are implemented unsigned 32 bits integers where 1 == true and 0 == false. It's just simpler to use that over an array of bool since you can reuse most of the buffers using inplace operations.